src/Arachne/Validation/File/ValidatorFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Arachne package.
+ *
+ * (c) Wojtek Gancarczyk <gancarczyk@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Arachne\Validation\File;
+
+use Arachne\Exception;
+use JsonSchema\RefResolver;
+use JsonSchema\Uri\UriRetriever;
+use JsonSchema\Validator;
+
+/**
+ * Class ValidatorFactory
+ * @package Arachne\Validation\Schema
+ * @author Jan Schädlich <schaedlich.jan@gmail.com>
+ */
+class ValidatorFactory
+{
+    /**
+     * @param $fileType
+     * @return JsonFile|XmlFile
+     */
+    public function create($fileType)
+    {
```

Comment:
By forcing the users to use the file extension in the scenario you break the backwards compatibility. My initial idea was to use a configuration variable to specify the data type returned by a specific web service under test. In my opinion it is pretty uncommon that the same web service will return two different data types, however. I like this implementation and I would keep it for now, but I would like you to make a small change to this factory and make it always return a `JsonFile` validator, if the file type was not recognized.